scripts/abi_artifact_generator/templates/functions.h.jinja_template

```diff
@@ -48,6 +48,22 @@ typedef struct {
 
 /*{ endfor -}*/
 
+// There's a slight difference in the way this pragma impacts the structs
+// between msvc and gcc, but it shouldn't be relevant in our case: since all
+// we care about is that these structs are emitted as "unnaturally aligned",
+// without any regard to their specific size.
```

Comment:
We don't care because we get the size as `sizeof(struct)` - so whatever compiler decides to do with it - it's consistent. And as long as it also considers said struct unnaturally aligned - we're happy.

Reply:
Yes, that's what the comment says to. But why we're happy? What happens with unaligned `struct`s downstream? Don't we want to test `struct` offsets?

Reply:
OK, please add to the comment the following part of your answer:

> When forcing them to be unaligned, compiler is definitely putting them on the stack (he doesn't want to deal with it either). So, in essence, these tests are only there so we can verify that function types layouts are aware that some stuff is bound to only use stack

Reply:
Done.